realm/realm-library/src/androidTest/kotlin/io/realm/ManagedSetTester.kt

```diff
@@ -439,15 +494,110 @@ fun managedSetFactory(): List<SetTester> {
     }
 
     // Create Mixed testers now
-    val mixedTesters = MixedType.values().map { mixedType ->
-        UnmanagedSetTester<Mixed>(
-                "UnmanagedSetMixed-${mixedType.name}",
-                getMixedValues(mixedType),
-                VALUE_MIXED_NOT_PRESENT
-        )
-    }
+    // TODO
 
-    // Put the together
+    // Put them together
 //    return primitiveTesters.plus(mixedTesters)
     return primitiveTesters
 }
+
+/**
+ * TODO
+ */
+abstract class ToArrayManaged<T> {
+
+    abstract fun toArrayWithParameter(realm: Realm, set: RealmSet<T>, values: List<T?>)
+
+    protected fun test(
+            realm: Realm,
+            set: RealmSet<T>,
+            values: List<T?>,
+            emptyArray: Array<T>,
+            fullArray: Array<T>
+    ) {
+        val emptyFromSet = set.toArray(emptyArray)
+        assertEquals(0, emptyFromSet.size)
+
+        realm.executeTransaction {
+            set.addAll(values as Collection<T>)
+        }
+        val fullFromSet = set.toArray(fullArray)
+        assertEquals(values.size, fullFromSet.size)
```

Comment:
Again, maybe worth a check for the actual values if it is just a matter of calling `assertArrayValues`

Reply:
Links to the comment above.